rpcs3/Emu/CPU/CPUTranslator.h

```diff
@@ -2283,9 +2283,11 @@ struct llvm_splat
 
 		if (auto i = llvm::dyn_cast_or_null<llvm::ShuffleVectorInst>(value))
 		{
-			if (llvm::isa<llvm::ConstantAggregateZero>(i->getOperand(2)))
+			if (llvm::isa<llvm::ConstantAggregateZero>(i->getOperand(1)) || llvm::isa<llvm::UndefValue>(i->getOperand(1)))
 			{
-				if (auto j = llvm::dyn_cast<llvm::InsertElementInst>(i->getOperand(0)))
+				static constexpr int zero_array[llvm_value_t<U>::is_vector]{};
```

Comment:
I think theres a typo here, definitely not a boolean to indicate array size, maybe vector elements count.

Reply:
is_vector is not boolean, it's a value that has vector number element or zero if it's not a vector. Confusing, but simplifies some things.